### PR TITLE
python311Packages.objgraph: 3.5.0 -> 3.6.0

### PR DESCRIPTION
**Pull request by fabaff**
Changelog: https://github.com/mgedmin/objgraph/blob/3.6.0/CHANGES.rst

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 240527` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>hackneyed</li>
    <li>python310Packages.mapbox</li>
    <li>python310Packages.mapbox.dist</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>frigate</li>
    <li>python310Packages.jaraco-abode</li>
    <li>python310Packages.jaraco-abode.dist</li>
  </ul>
</details>
<details>
  <summary>101 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>asciidoc-full-with-plugins.dist</li>
    <li>asciidoc-full.dist</li>
    <li>ayu-theme-gtk</li>
    <li>capitaine-cursors</li>
    <li>ceph (ceph-dev)</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph.dev (ceph-dev.dev)</li>
    <li>ceph.doc (ceph-dev.doc)</li>
    <li>libceph (ceph.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph.man (ceph-dev.man)</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>cinnamon.iso-flags-png-320x420</li>
    <li>cinnamon.iso-flags-svg</li>
    <li>clevis</li>
    <li>clevis.man</li>
    <li>coreboot-configurator</li>
    <li>cwltool</li>
    <li>cwltool.dist</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>emojione</li>
    <li>fim</li>
    <li>flexget</li>
    <li>flexget.dist</li>
    <li>hikounomizu</li>
    <li>home-assistant-component-tests.profiler</li>
    <li>inkscape</li>
    <li>inkscape-with-extensions</li>
    <li>iso-flags</li>
    <li>k40-whisperer</li>
    <li>kabeljau</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>luksmeta</li>
    <li>mate.mate-utils</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mojave-gtk-theme</li>
    <li>numix-cursor-theme</li>
    <li>numix-solarized-gtk-theme</li>
    <li>pdm</li>
    <li>pdm.dist</li>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
    <li>plata-theme</li>
    <li>poetry</li>
    <li>poetry.dist</li>
    <li>poetryPlugins.poetry-audit-plugin</li>
    <li>poetryPlugins.poetry-audit-plugin.dist</li>
    <li>poetryPlugins.poetry-plugin-up</li>
    <li>poetryPlugins.poetry-plugin-up.dist</li>
    <li>pop-gtk-theme</li>
    <li>python310Packages.cachecontrol</li>
    <li>python310Packages.cachecontrol.dist</li>
    <li>python310Packages.cacheyou</li>
    <li>python310Packages.cacheyou.dist</li>
    <li>python310Packages.cherrypy</li>
    <li>python310Packages.cherrypy.dist</li>
    <li>python310Packages.cwl-upgrader</li>
    <li>python310Packages.cwl-upgrader.dist</li>
    <li>python310Packages.cwl-utils</li>
    <li>python310Packages.cwl-utils.dist</li>
    <li>python310Packages.diagrams</li>
    <li>python310Packages.diagrams.dist</li>
    <li>python310Packages.jaraco-net</li>
    <li>python310Packages.jaraco-net.dist</li>
    <li>python310Packages.objgraph</li>
    <li>python310Packages.objgraph.dist</li>
    <li>python310Packages.pyalgotrade</li>
    <li>python310Packages.pyalgotrade.dist</li>
    <li>python310Packages.pylxd</li>
    <li>python310Packages.pylxd.dist</li>
    <li>python310Packages.schema-salad</li>
    <li>python310Packages.schema-salad.dist</li>
    <li>python310Packages.scikit-bio</li>
    <li>python310Packages.scikit-bio.dist</li>
    <li>python310Packages.ws4py</li>
    <li>python310Packages.ws4py.dist</li>
    <li>python311Packages.diagrams</li>
    <li>python311Packages.diagrams.dist</li>
    <li>python311Packages.objgraph</li>
    <li>python311Packages.objgraph.dist</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>sabnzbd</li>
    <li>sambaFull (samba4Full)</li>
    <li>sambaFull.dev (samba4Full.dev)</li>
    <li>sambaFull.man (samba4Full.man)</li>
    <li>tang</li>
    <li>tang.man</li>
    <li>tebreak</li>
    <li>tebreak.dist</li>
    <li>tribler</li>
    <li>volantes-cursors</li>
    <li>wpa_supplicant_gui</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 240527` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>hackneyed</li>
    <li>python310Packages.mapbox</li>
    <li>python310Packages.mapbox.dist</li>
    <li>spring</li>
    <li>springLobby</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>frigate</li>
  </ul>
</details>
<details>
  <summary>103 packages built:</summary>
  <ul>
    <li>adapta-gtk-theme</li>
    <li>arc-theme</li>
    <li>arx-libertatis</li>
    <li>asciidoc-full</li>
    <li>asciidoc-full-with-plugins</li>
    <li>asciidoc-full-with-plugins.dist</li>
    <li>asciidoc-full.dist</li>
    <li>ayu-theme-gtk</li>
    <li>capitaine-cursors</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph-dev</li>
    <li>ceph-dev.dev</li>
    <li>ceph-dev.doc</li>
    <li>libceph (ceph-dev.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph-dev.man</li>
    <li>cinnamon.cinnamon-screensaver</li>
    <li>cinnamon.iso-flags-png-320x420</li>
    <li>cinnamon.iso-flags-svg</li>
    <li>clevis</li>
    <li>clevis.man</li>
    <li>coreboot-configurator</li>
    <li>cwltool</li>
    <li>cwltool.dist</li>
    <li>dblatexFull</li>
    <li>disorderfs</li>
    <li>emojione</li>
    <li>fim</li>
    <li>flexget</li>
    <li>flexget.dist</li>
    <li>hikounomizu</li>
    <li>home-assistant-component-tests.profiler</li>
    <li>inkscape</li>
    <li>inkscape-with-extensions</li>
    <li>iso-flags</li>
    <li>k40-whisperer</li>
    <li>kabeljau</li>
    <li>rep (kakounePlugins.rep)</li>
    <li>luksmeta</li>
    <li>mate.mate-utils</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mojave-gtk-theme</li>
    <li>numix-cursor-theme</li>
    <li>numix-solarized-gtk-theme</li>
    <li>pdm</li>
    <li>pdm.dist</li>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
    <li>plata-theme</li>
    <li>poetry</li>
    <li>poetry.dist</li>
    <li>poetryPlugins.poetry-audit-plugin</li>
    <li>poetryPlugins.poetry-audit-plugin.dist</li>
    <li>poetryPlugins.poetry-plugin-up</li>
    <li>poetryPlugins.poetry-plugin-up.dist</li>
    <li>pop-gtk-theme</li>
    <li>python310Packages.cachecontrol</li>
    <li>python310Packages.cachecontrol.dist</li>
    <li>python310Packages.cacheyou</li>
    <li>python310Packages.cacheyou.dist</li>
    <li>python310Packages.cherrypy</li>
    <li>python310Packages.cherrypy.dist</li>
    <li>python310Packages.cwl-upgrader</li>
    <li>python310Packages.cwl-upgrader.dist</li>
    <li>python310Packages.cwl-utils</li>
    <li>python310Packages.cwl-utils.dist</li>
    <li>python310Packages.diagrams</li>
    <li>python310Packages.diagrams.dist</li>
    <li>python310Packages.jaraco-abode</li>
    <li>python310Packages.jaraco-abode.dist</li>
    <li>python310Packages.jaraco-net</li>
    <li>python310Packages.jaraco-net.dist</li>
    <li>python310Packages.objgraph</li>
    <li>python310Packages.objgraph.dist</li>
    <li>python310Packages.pyalgotrade</li>
    <li>python310Packages.pyalgotrade.dist</li>
    <li>python310Packages.pylxd</li>
    <li>python310Packages.pylxd.dist</li>
    <li>python310Packages.schema-salad</li>
    <li>python310Packages.schema-salad.dist</li>
    <li>python310Packages.scikit-bio</li>
    <li>python310Packages.scikit-bio.dist</li>
    <li>python310Packages.ws4py</li>
    <li>python310Packages.ws4py.dist</li>
    <li>python311Packages.diagrams</li>
    <li>python311Packages.diagrams.dist</li>
    <li>python311Packages.objgraph</li>
    <li>python311Packages.objgraph.dist</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>sabnzbd</li>
    <li>sambaFull (samba4Full)</li>
    <li>sambaFull.dev (samba4Full.dev)</li>
    <li>sambaFull.man (samba4Full.man)</li>
    <li>tang</li>
    <li>tang.man</li>
    <li>tebreak</li>
    <li>tebreak.dist</li>
    <li>tribler</li>
    <li>volantes-cursors</li>
    <li>wpa_supplicant_gui</li>
  </ul>
</details>

**Comment by fabaff** [comment]
``frigate is already broken.

File: pkgs/development/python-modules/objgraph/default.nix [hunk 1]
```diff
@@ -11,14 +11,14 @@
 
 buildPythonPackage rec {
   pname = "objgraph";
-  version = "3.5.0";
+  version = "3.6.0";
   format = "setuptools";
 
   disabled = pythonOlder "3.5" || isPyPy;
```
Comment (dotlambda) [inline pkgs/development/python-modules/objgraph/default.nix:17]:
```suggestion
  disabled = pythonOlder "3.7" || isPyPy;
```

File: pkgs/development/python-modules/jaraco-abode/default.nix [hunk 2]
```diff
@@ -83,10 +84,12 @@ buildPythonPackage rec {
     "test_cookies"
     "test_empty_cookies"
     "test_invalid_cookies"
+    # Issue with the regex
+    "test_camera_capture_no_control_URLs"
   ];
 
   meta = with lib; {
-    changelog = "https://github.com/jaraco/jaraco.abode/blob/${src.rev}/CHANGES.rst";
+    changelog = "https://github.com/jaraco/jaraco.abode/blob/${version}/CHANGES.rst";
```
Comment (dotlambda) [inline pkgs/development/python-modules/jaraco-abode/default.nix:92]: 404: https://github.com/jaraco/jaraco.abode/blob/5.1.0/CHANGES.rst

Comment (dotlambda) [reply pkgs/development/python-modules/jaraco-abode/default.nix:92]:
```sugestion
    changelog = "https://github.com/jaraco/jaraco.abode/blob/${src.rev}/NEWS.rst";
```

File: pkgs/development/python-modules/jaraco-abode/default.nix [hunk 3]
```diff
@@ -1,41 +1,40 @@
...
+, setuptools-scm
 }:
 
 buildPythonPackage rec {
   pname = "jaraco-abode";
-  version = "5.0.1";
+  version = "5.1.0";
+  format = "pyproject";
 
   disabled = pythonOlder "3.7";
```
Comment (dotlambda) [inline pkgs/development/python-modules/jaraco-abode/default.nix:31]:
```suggestion
  disabled = pythonOlder "3.8";
```

File: pkgs/development/python-modules/objgraph/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/development/python-modules/objgraph/default.nix:17]: I fixed that in https://github.com/NixOS/nixpkgs/pull/244135.

**Pull request merged**